### PR TITLE
User activity guide

### PR DESCRIPTION
**Pull request by ddworak**

### CODE REVIEW

File: guide/src/main/scala/io/udash/web/guide/views/ext/UserActivityExtView.scala [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    p(
+      """When it comes to website tracking there are a plethora of metrics at our disposal.
+        |If we’re talking user engagement, we might look at the bounce rate, average time on site, or average page views per visit.
+        |For organic search, we might look at the number of organic visits, top organic keywords, and others.
+        |Then there’s content — the king of all data. Which web pages are the most popular?
+        |What are the most common navigation flows? Which features are most commonly used?
+        |Udash user activity extenstions enable you to gather the data you need to provide the best user experience for your website.""".stripMargin
+    ),
+    h2("Browser navigation"),
+    p("To enable browser navigation tracking, simply mixin UrlLogging into your frontend application.",
```
Comment (Starzu) [inline guide/src/main/scala/io/udash/web/guide/views/ext/UserActivityExtView.scala]: Space missing at the end of the string,

File: guide/src/main/scala/io/udash/web/guide/views/ext/UserActivityExtView.scala [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+      "The ", i("log(url, referrer)"), " method will be called whenever the user changes app state."),
+    p("You can see this mechanism in action here in the guide. We've already provided the implementation:"),
+    CodeBlock(
+      s"""implicit val applicationInstance = new Application[RoutingState](routingRegistry, viewPresenterRegistry, RootState) with UrlLogging[RoutingState] {
+          |    override protected def log(url: String, referrer: Option[String]): Unit = UrlLoggingDemo.log(url, referrer)
+          |}""".stripMargin
```
Comment (Starzu) [inline guide/src/main/scala/io/udash/web/guide/views/ext/UserActivityExtView.scala:68]: Try to make these lines shorter, please. :)

File: guide/src/main/scala/io/udash/web/guide/views/ext/UserActivityExtView.scala [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    h2("RPC call logging"),
+    p("Enabling backend call logging is also quite simple. In order to define logging behaviour, you have to mix ",
+      "CallLogging into your ExposesServerRPC, e.g.:"),
+    CodeBlock(
+      """ new DefaultExposesServerRPC[MainServerRPC](new ExposedRpcInterfaces(clientId)) with CallLogging[MainServerRPC] {
+        |  override protected val metadata: RPCMetadata[MainServerRPC] = RPCMetadata[MainServerRPC]
+        |
+        |  override def log(rpcName: String, methodName: String, args: Seq[String]): Unit =
+        |    println(s"$rpcName $methodName $args")
+        |} """.stripMargin)(GuideStyles),
```
Comment (Starzu) [inline guide/src/main/scala/io/udash/web/guide/views/ext/UserActivityExtView.scala:96]: Same here - too long lines.

**Comment by Starzu** [comment]
This version does not compile with the latest Udash `0.3.0-SNAPSHOT`. 

```
ApplicationServer.scala:28: not enough arguments for constructor ExposedRpcInterfaces: (callLogger: io.udash.web.guide.demos.activity.CallLogger)(implicit clientId: io.udash.rpc.ClientId)io.
udash.web.guide.rpc.ExposedRpcInterfaces.
[error] Unspecified value parameter callLogger.
[error]       new DefaultExposesServerRPC[MainServerRPC](new ExposedRpcInterfaces()(clientId))
```

You should also provide selenium tests for these live demos.

**Comment by ddworak** [comment]
Compilation error was my bad file selection for commit, should work fine now.
I don't see anything in /ext tested by Selenium - why was https://github.com/UdashFramework/udash-guide/pull/7 merged?

**Comment by Starzu** [comment]
There are selenium tests for jQuery ext: 7c3e025a96f4dd0ad68bd9e86066abf9fd1d89b9
That's right, tests for i18n are missing. I'm going to add them.

File: guide/src/main/scala/io/udash/web/guide/views/ext/UserActivityExtView.scala [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+          |  val enabled = Property(false)
+          |  val history = SeqProperty[(String, Option[String])](ListBuffer.empty)
+          |  enabled.listen(b => if(!b) history.set(ListBuffer.empty))
+          |
+          |  def log(url: String, referrer: Option[String]): Unit =
+          |    if(enabled.get) history.append((url, referrer))
+          |}""".stripMargin
+    )(GuideStyles),
+    p("to see it in action just enable logging below, switch to another chapter and come back here."), br,
+    span(GuideStyles.frame)(
```
Comment (Starzu) [inline guide/src/main/scala/io/udash/web/guide/views/ext/UserActivityExtView.scala]: Sorry, I didn't notice this earlier, but could you separate this demo component into `io.udash.web.guide.views.ext.demo` package? Take a look at `DynamicRemoteTranslationsDemo` for example.

File: guide/src/main/scala/io/udash/web/guide/views/ext/UserActivityExtView.scala [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+        |import io.udash.rpc.utils.Logged
+        |
+        |@RPC
+        |trait PingPongServerRPC {
+        |
+        |  @Logged
+        |  def fPing(id: Int): Future[Int]
+        |}""".stripMargin)(GuideStyles),
+    new PingPongCallDemoComponent,
+    span(GuideStyles.frame)(
```
Comment (Starzu) [inline guide/src/main/scala/io/udash/web/guide/views/ext/UserActivityExtView.scala]: This one too, please. :)

**Comment by Starzu** [comment]
@ddworak take a look at my suggestions, please

**Comment by ddworak** [comment]
:yum: :ship:

**Pull request merged**